manpages/brew.1
.\" generated with Ronn/v0.7.3
.\" http://github.com/rtomayko/ronn/tree/0.7.3
.
.TH "BREW" "1" "June 2021" "Homebrew" "brew"
.
.SH "NAME"
\fBbrew\fR \- The Missing Package Manager for macOS (or Linux)
.
.SH "SYNOPSIS"
\fBbrew\fR \fB\-\-version\fR
.
.br
\fBbrew\fR \fIcommand\fR [\fB\-\-verbose\fR|\fB\-v\fR] [\fIoptions\fR] [\fIformula\fR] \.\.\.
.
.SH "DESCRIPTION"
Homebrew is the easiest and most flexible way to install the UNIX tools Apple didn\'t include with macOS\. It can also install software not packaged for your Linux distribution to your home directory without requiring \fBsudo\fR\.
.
.SH "TERMINOLOGY"
\fBformula\fR
    Homebrew package definition built from upstream sources
.
.P
\fBcask\fR
    Homebrew package definition that installs macOS native applications
.
.P
\fBkeg\fR
    installation destination directory of a given \fBformula\fR version e\.g\. \fB/usr/local/Cellar/foo/0\.1\fR
.
.P
\fBrack\fR
    directory containing one or more versioned kegs e\.g\. \fB/usr/local/Cellar/foo\fR
.
.P
\fBkeg\-only\fR
    a \fBformula\fR is \fBkeg\-only\fR if it is not symlinked into Homebrew\'s prefix (e\.g\. \fB/usr/local\fR)
.
.P
\fBcellar\fR
    directory containing one or more named \fBracks\fR e\.g\. \fB/usr/local/Cellar\fR
.
.P
\fBCaskroom\fR
    directory containing one or more named \fBcasks\fR e\.g\. \fB/usr/local/Caskroom\fR
.
.P
\fBexternal command\fR
    \fBbrew\fR subcommand defined outside of the Homebrew/brew GitHub repository
.
.P
\fBtap\fR
    directory (and usually Git repository) of \fBformulae\fR, \fBcasks\fR and/or \fBexternal commands\fR
.
.P
\fBbottle\fR
    pre\-built \fBkeg\fR poured into the \fBcellar\fR/\fBrack\fR instead of building from upstream sources
.
.SH "ESSENTIAL COMMANDS"
For the full command list, see the \fICOMMANDS\fR section\.
.
.P
With \fB\-\-verbose\fR or \fB\-\-debug\fR, many commands print extra debugging information\. Note that these options should only appear after a command\.
.
.SS "\fBinstall\fR \fIformula\fR"
Install \fIformula\fR\.
.
.P
\fIformula\fR is usually the name of the formula to install, but it has other syntaxes which are listed in the \fISPECIFYING FORMULAE\fR section\.
.
.SS "\fBuninstall\fR \fIformula\fR"
Uninstall \fIformula\fR\.
.
.SS "\fBlist\fR"
List all installed formulae\.
.
.SS "\fBsearch\fR [\fItext\fR|\fB/\fR\fItext\fR\fB/\fR]"
Perform a substring search of cask tokens and formula names for \fItext\fR\. If \fItext\fR is flanked by slashes, it is interpreted as a regular expression\. The search for \fItext\fR is extended online to \fBhomebrew/core\fR and \fBhomebrew/cask\fR\. If no search term is provided, all locally available formulae are listed\.
.
.SH "COMMANDS"
.
.SS "\fBanalytics\fR [\fIsubcommand\fR]"
Control Homebrew\'s anonymous aggregate user behaviour analytics\. Read more at \fIhttps://docs\.brew\.sh/Analytics\fR\.
.
.P
\fBbrew analytics\fR [\fBstate\fR]
    Display the current state of Homebrew\'s analytics\.
.
.P
\fBbrew analytics\fR (\fBon\fR|\fBoff\fR)
    Turn Homebrew\'s analytics on or off respectively\.
.
.P
\fBbrew analytics regenerate\-uuid\fR
    Regenerate the UUID used for Homebrew\'s analytics\.
.
.SS "\fBautoremove\fR [\fI\-\-dry\-run\fR]"
Uninstall formulae that were only installed as a dependency of another formula and are now no longer needed\.
.
.TP
\fB\-n\fR, \fB\-\-dry\-run\fR
List what would be uninstalled, but do not actually uninstall anything\.
.
.SS "\fBcasks\fR"
List all locally installable casks including short names\.
.
.SS "\fBcleanup\fR [\fIoptions\fR] [\fIformula\fR|\fIcask\fR \.\.\.]"
Remove stale lock files and outdated downloads for all formulae and casks, and remove old versions of installed formulae\. If arguments are specified, only do this for the given formulae and casks\. Removes all downloads more than 120 days old\. This can be adjusted with \fBHOMEBREW_CLEANUP_MAX_AGE_DAYS\fR\.
.
.TP
\fB\-\-prune\fR
Remove all cache files older than specified \fIdays\fR\. If you want to remove everything, use \fB\-\-prune=all\fR\.
.
.TP
\fB\-n\fR, \fB\-\-dry\-run\fR
Show what would be removed, but do not actually remove anything\.
.
.TP
\fB\-s\fR
Scrub the cache, including downloads for even the latest versions\. Note downloads for any installed formulae or casks will still not be deleted\. If you want to delete those too: \fBrm \-rf "$(brew \-\-cache)"\fR
.
.TP
\fB\-\-prune\-prefix\fR
Only prune the symlinks and directories from the prefix and remove no other files\.
.
.SS "\fBcommands\fR [\fI\-\-quiet\fR] [\fI\-\-include\-aliases\fR]"
Show lists of built\-in and external commands\.
.
.TP
\fB\-q\fR, \fB\-\-quiet\fR
List only the names of commands without category headers\.
.
.TP
\fB\-\-include\-aliases\fR
Include aliases of internal commands\.
.
.SS "\fBcompletions\fR [\fIsubcommand\fR]"
Control whether Homebrew automatically links external tap shell completion files\. Read more at \fIhttps://docs\.brew\.sh/Shell\-Completion\fR\.
.
.P
\fBbrew completions\fR [\fBstate\fR]
    Display the current state of Homebrew\'s completions\.
.
.P
\fBbrew completions\fR (\fBlink\fR|\fBunlink\fR)
    Link or unlink Homebrew\'s completions\.
.
.SS "\fBconfig\fR, \fB\-\-config\fR"
Show Homebrew and system configuration info useful for debugging\. If you file a bug report, you will be required to provide this information\.
.
.SS "\fBdeps\fR [\fIoptions\fR] [\fIformula\fR|\fIcask\fR \.\.\.]"
Show dependencies for \fIformula\fR\. Additional options specific to \fIformula\fR may be appended to the command\. When given multiple formula arguments, show the intersection of dependencies for each formula\.
.
.TP
\fB\-n\fR
Sort dependencies in topological order\.
.
.TP
\fB\-\-1\fR
Only show dependencies one level down, instead of recursing\.
.
.TP
\fB\-\-union\fR
Show the union of dependencies for multiple \fIformula\fR, instead of the intersection\.
.
.TP
\fB\-\-full\-name\fR
List dependencies by their full name\.
.
.TP
\fB\-\-include\-build\fR
Include \fB:build\fR dependencies for \fIformula\fR\.
.
.TP
\fB\-\-include\-optional\fR
Include \fB:optional\fR dependencies for \fIformula\fR\.
.
.TP
\fB\-\-include\-test\fR
Include \fB:test\fR dependencies for \fIformula\fR (non\-recursive)\.
.
.TP
\fB\-\-skip\-recommended\fR
Skip \fB:recommended\fR dependencies for \fIformula\fR\.
.
.TP
\fB\-\-include\-requirements\fR
Include requirements in addition to dependencies for \fIformula\fR\.
.
.TP
\fB\-\-tree\fR
Show dependencies as a tree\. When given multiple formula arguments, show individual trees for each formula\.
.
.TP
\fB\-\-annotate\fR
Mark any build, test, optional, or recommended dependencies as such in the output\.
.
.TP
\fB\-\-installed\fR
List dependencies for formulae that are currently installed\. If \fIformula\fR is specified, list only its dependencies that are currently installed\.
.
.TP
\fB\-\-all\fR
List dependencies for all available formulae\.
.
.TP
\fB\-\-for\-each\fR
Switch into the mode used by the \fB\-\-all\fR option, but only list dependencies for each provided \fIformula\fR, one formula per line\. This is used for debugging the \fB\-\-installed\fR/\fB\-\-all\fR display mode\.
.
.TP
\fB\-\-formula\fR
Treat all named arguments as formulae\.
.
.TP
\fB\-\-cask\fR
Treat all named arguments as casks\.
.
.SS "\fBdesc\fR [\fIoptions\fR] \fIformula\fR|\fItext\fR|\fB/\fR\fIregex\fR\fB/\fR [\.\.\.]"
Display \fIformula\fR\'s name and one\-line description\. Formula descriptions are cached; the cache is created on the first search, making that search slower than subsequent ones\.
.
.TP
\fB\-s\fR, \fB\-\-search\fR
Search both names and descriptions for \fItext\fR\. If \fItext\fR is flanked by slashes, it is interpreted as a regular expression\.
.
.TP
\fB\-n\fR, \fB\-\-name\fR
Search just names for \fItext\fR\. If \fItext\fR is flanked by slashes, it is interpreted as a regular expression\.
.
.TP
\fB\-d\fR, \fB\-\-description\fR
Search just descriptions for \fItext\fR\. If \fItext\fR is flanked by slashes, it is interpreted as a regular expression\.
.
.SS "\fBdoctor\fR, \fBdr\fR [\fI\-\-list\-checks\fR] [\fI\-\-audit\-debug\fR] [\fIdiagnostic_check\fR \.\.\.]"
Check your system for potential problems\. Will exit with a non\-zero status if any potential problems are found\. Please note that these warnings are just used to help the Homebrew maintainers with debugging if you file an issue\. If everything you use Homebrew for is working fine: please don\'t worry or file an issue; just ignore this\.
.
.TP
\fB\-\-list\-checks\fR
List all audit methods, which can be run individually if provided as arguments\.
.
.TP
\fB\-D\fR, \fB\-\-audit\-debug\fR
Enable debugging and profiling of audit methods\.
.
.SS "\fBfetch\fR [\fIoptions\fR] \fIformula\fR|\fIcask\fR [\.\.\.]"
Download a bottle (if available) or source packages for \fIformula\fRe and binaries for \fIcask\fRs\. For files, also print SHA\-256 checksums\.
.
.TP
\fB\-\-HEAD\fR
Fetch HEAD version instead of stable version\.
.
.TP
\fB\-f\fR, \fB\-\-force\fR
Remove a previously cached version and re\-fetch\.
.
.TP
\fB\-v\fR, \fB\-\-verbose\fR
Do a verbose VCS checkout, if the URL represents a VCS\. This is useful for seeing if an existing VCS cache has been updated\.
.
.TP
\fB\-\-retry\fR
Retry if downloading fails or re\-download if the checksum of a previously cached version no longer matches\.
.
.TP
\fB\-\-deps\fR
Also download dependencies for any listed \fIformula\fR\.
.
.TP
\fB\-s\fR, \fB\-\-build\-from\-source\fR
Download source packages rather than a bottle\.
.
.TP
\fB\-\-build\-bottle\fR
Download source packages (for eventual bottling) rather than a bottle\.
.
.TP
\fB\-\-force\-bottle\fR
Download a bottle if it exists for the current or newest version of macOS, even if it would not be used during installation\.
.
.TP
\fB\-\-[no\-]quarantine\fR
Disable/enable quarantining of downloads (default: enabled)\.
.
.TP
\fB\-\-formula\fR
Treat all named arguments as formulae\.
.
.TP
\fB\-\-cask\fR
Treat all named arguments as casks\.
.
.SS "\fBformulae\fR"
List all locally installable formulae including short names\.
.
.SS "\fBgist\-logs\fR [\fIoptions\fR] \fIformula\fR"
Upload logs for a failed build of \fIformula\fR to a new Gist\. Presents an error message if no logs are found\.
.
.TP
\fB\-\-with\-hostname\fR
Include the hostname in the Gist\.
.
.TP
\fB\-n\fR, \fB\-\-new\-issue\fR
Automatically create a new issue in the appropriate GitHub repository after creating the Gist\.
.
.TP
\fB\-p\fR, \fB\-\-private\fR
The Gist will be marked private and will not appear in listings but will be accessible with its link\.
.
.SS "\fBhome\fR, \fBhomepage\fR [\fI\-\-formula\fR] [\fI\-\-cask\fR] [\fIformula\fR|\fIcask\fR \.\.\.]"
Open a \fIformula\fR or \fIcask\fR\'s homepage in a browser, or open Homebrew\'s own homepage if no argument is provided\.
.
.TP
\fB\-\-formula\fR
Treat all named arguments as formulae\.
.
.TP
\fB\-\-cask\fR
Treat all named arguments as casks\.
.
.SS "\fBinfo\fR, \fBabv\fR [\fIoptions\fR] [\fIformula\fR|\fIcask\fR \.\.\.]"
Display brief statistics for your Homebrew installation\.
.
.P
If a \fIformula\fR or \fIcask\fR is provided, show summary of information about it\.
.
.TP
\fB\-\-analytics\fR
List global Homebrew analytics data or, if specified, installation and build error data for \fIformula\fR (provided neither \fBHOMEBREW_NO_ANALYTICS\fR nor \fBHOMEBREW_NO_GITHUB_API\fR are set)\.
.
.TP
\fB\-\-days\fR
How many days of analytics data to retrieve\. The value for \fIdays\fR must be \fB30\fR, \fB90\fR or \fB365\fR\. The default is \fB30\fR\.
.
.TP
\fB\-\-category\fR
Which type of analytics data to retrieve\. The value for \fIcategory\fR must be \fBinstall\fR, \fBinstall\-on\-request\fR or \fBbuild\-error\fR; \fBcask\-install\fR or \fBos\-version\fR may be specified if \fIformula\fR is not\. The default is \fBinstall\fR\.
.
.TP
\fB\-\-github\fR
Open the GitHub source page for \fIformula\fR in a browser\. To view formula history locally: \fBbrew log \-p\fR \fIformula\fR
.
.TP
\fB\-\-json\fR
Print a JSON representation\. Currently the default value for \fIversion\fR is \fBv1\fR for \fIformula\fR\. For \fIformula\fR and \fIcask\fR use \fBv2\fR\. See the docs for examples of using the JSON output: \fIhttps://docs\.brew\.sh/Querying\-Brew\fR
.
.TP
\fB\-\-installed\fR
Print JSON of formulae that are currently installed\.
.
.TP
\fB\-\-all\fR
Print JSON of all available formulae\.
.
.TP
\fB\-v\fR, \fB\-\-verbose\fR
Show more verbose analytics data for \fIformula\fR\.
.
.TP
\fB\-\-formula\fR
Treat all named arguments as formulae\.
.
.TP
\fB\-\-cask\fR
Treat all named arguments as casks\.
.
.SS "\fBinstall\fR [\fIoptions\fR] \fIformula\fR|\fIcask\fR [\.\.\.]"
Install a \fIformula\fR or \fIcask\fR\. Additional options specific to a \fIformula\fR may be appended to the command\.
.
.P
If \fBHOMEBREW_BUILD_FROM_SOURCE\fR is set, then both \fIformula\fR and any dependencies installed while installing it are built from source even if bottles are available for them, regardless of whether \fB\-\-build\-from\-source\fR was passed or not\.
.
.P
Unless \fBHOMEBREW_NO_INSTALL_CLEANUP\fR is set, \fBbrew cleanup\fR will then be run for the installed formulae or, every 30 days, for all formulae\.
.
.TP
\fB\-d\fR, \fB\-\-debug\fR
If brewing fails, open an interactive debugging session with access to IRB or a shell inside the temporary build directory\.
.
.TP
\fB\-f\fR, \fB\-\-force\fR
Install formulae without checking for previously installed keg\-only or non\-migrated versions\. When installing casks, overwrite existing files (binaries and symlinks are excluded, unless originally from the same cask)\.
.
.TP
\fB\-v\fR, \fB\-\-verbose\fR
Print the verification and postinstall steps\.
.
.TP
\fB\-\-formula\fR
Treat all named arguments as formulae\.
.
.TP
\fB\-\-env\fR
Disabled other than for internal Homebrew use\.
.
.TP
\fB\-\-ignore\-dependencies\fR
An unsupported Homebrew development flag to skip installing any dependencies of any kind\. If the dependencies are not already present, the formula will have issues\. If you\'re not developing Homebrew, consider adjusting your PATH rather than using this flag\.
.
.TP
\fB\-\-only\-dependencies\fR
Install the dependencies with specified options but do not install the formula itself\.
.
.TP
\fB\-\-cc\fR
Attempt to compile using the specified \fIcompiler\fR, which should be the name of the compiler\'s executable, e\.g\. \fBgcc\-7\fR for GCC 7\. In order to use LLVM\'s clang, specify \fBllvm_clang\fR\. To use the Apple\-provided clang, specify \fBclang\fR\. This option will only accept compilers that are provided by Homebrew or bundled with macOS\. Please do not file issues if you encounter errors while using this option\.
.
.TP
\fB\-s\fR, \fB\-\-build\-from\-source\fR
Compile \fIformula\fR from source even if a bottle is provided\. Dependencies will still be installed from bottles if they are available\.
.
.TP
\fB\-\-force\-bottle\fR
Install from a bottle if it exists for the current or newest version of macOS, even if it would not normally be used for installation\.
.
.TP
\fB\-\-include\-test\fR
Install testing dependencies required to run \fBbrew test\fR \fIformula\fR\.
.
.TP
\fB\-\-HEAD\fR
If \fIformula\fR defines it, install the HEAD version, aka\. main, trunk, unstable, master\.
.
.TP
\fB\-\-fetch\-HEAD\fR
Fetch the upstream repository to detect if the HEAD installation of the formula is outdated\. Otherwise, the repository\'s HEAD will only be checked for updates when a new stable or development version has been released\.
.
.TP
\fB\-\-keep\-tmp\fR
Retain the temporary files created during installation\.
.
.TP
\fB\-\-build\-bottle\fR
Prepare the formula for eventual bottling during installation, skipping any post\-install steps\.
.
.TP
\fB\-\-bottle\-arch\fR
Optimise bottles for the specified architecture rather than the oldest architecture supported by the version of macOS the bottles are built on\.
.
.TP
\fB\-\-display\-times\fR
Print install times for each formula at the end of the run\.
.
.TP
\fB\-i\fR, \fB\-\-interactive\fR
Download and patch \fIformula\fR, then open a shell\. This allows the user to run \fB\./configure \-\-help\fR and otherwise determine how to turn the software package into a Homebrew package\.
.
.TP
\fB\-g\fR, \fB\-\-git\fR
Create a Git repository, useful for creating patches to the software\.
.
.TP
\fB\-\-cask\fR
Treat all named arguments as casks\.
.
.TP
\fB\-\-[no\-]binaries\fR
Disable/enable linking of helper executables (default: enabled)\.
.
.TP
\fB\-\-require\-sha\fR
Require all casks to have a checksum\.
.
.TP
\fB\-\-[no\-]quarantine\fR
Disable/enable quarantining of downloads (default: enabled)\.
.
.TP
\fB\-\-skip\-cask\-deps\fR
Skip installing cask dependencies\.
.
.SS "\fBleaves\fR [\fI\-\-installed\-on\-request\fR] [\fI\-\-installed\-as\-dependency\fR]"
List installed formulae that are not dependencies of another installed formula\.
.
.TP
\fB\-r\fR, \fB\-\-installed\-on\-request\fR
Only list leaves that were manually installed\.
.
.TP
\fB\-p\fR, \fB\-\-installed\-as\-dependency\fR
Only list leaves that were installed as dependencies\.
.
.SS "\fBlink\fR, \fBln\fR [\fIoptions\fR] \fIinstalled_formula\fR [\.\.\.]"
Symlink all of \fIformula\fR\'s installed files into Homebrew\'s prefix\. This is done automatically when you install formulae but can be useful for DIY installations\.
.
.TP
\fB\-\-overwrite\fR
Delete files that already exist in the prefix while linking\.
.
.TP
\fB\-n\fR, \fB\-\-dry\-run\fR
List files which would be linked or deleted by \fBbrew link \-\-overwrite\fR without actually linking or deleting any files\.
.
.TP
\fB\-f\fR, \fB\-\-force\fR
Allow keg\-only formulae to be linked\.
.
.TP
\fB\-\-HEAD\fR
Link the HEAD version of the formula if it is installed\.
.
.SS "\fBlist\fR, \fBls\fR [\fIoptions\fR] [\fIinstalled_formula\fR|\fIinstalled_cask\fR \.\.\.]"
List all installed formulae and casks\.
.
.P
If \fIformula\fR is provided, summarise the paths within its current keg\. If \fIcask\fR is provided, list its artifacts\.
.
.TP
\fB\-\-formula\fR
List only formulae, or treat all named arguments as formulae\.
.
.TP
\fB\-\-cask\fR
List only casks, or treat all named arguments as casks\.
.
.TP
\fB\-\-full\-name\fR
Print formulae with fully\-qualified names\. Unless \fB\-\-full\-name\fR, \fB\-\-versions\fR or \fB\-\-pinned\fR are passed, other options (i\.e\. \fB\-1\fR, \fB\-l\fR, \fB\-r\fR and \fB\-t\fR) are passed to \fBls\fR(1) which produces the actual output\.
.
.TP
\fB\-\-versions\fR
Show the version number for installed formulae, or only the specified formulae if \fIformula\fR are provided\.
.
.TP
\fB\-\-multiple\fR
Only show formulae with multiple versions installed\.
.
.TP
\fB\-\-pinned\fR
List only pinned formulae, or only the specified (pinned) formulae if \fIformula\fR are provided\. See also \fBpin\fR, \fBunpin\fR\.
.
.TP
\fB\-1\fR
Force output to be one entry per line\. This is the default when output is not to a terminal\.
.
.TP
\fB\-l\fR
List formulae and/or casks in long format\. Has no effect when a formula or cask name is passed as an argument\.
.
.TP
\fB\-r\fR
Reverse the order of the formulae and/or casks sort to list the oldest entries first\. Has no effect when a formula or cask name is passed as an argument\.
.
.TP
\fB\-t\fR
Sort formulae and/or casks by time modified, listing most recently modified first\. Has no effect when a formula or cask name is passed as an argument\.
.
.SS "\fBlog\fR [\fIoptions\fR] [\fIformula\fR]"
Show the \fBgit log\fR for \fIformula\fR, or show the log for the Homebrew repository if no formula is provided\.
.
.TP
\fB\-p\fR, \fB\-\-patch\fR
Also print patch from commit\.
.
.TP
\fB\-\-stat\fR
Also print diffstat from commit\.
.
.TP
\fB\-\-oneline\fR
Print only one line per commit\.
.
.TP
\fB\-1\fR
Print only one commit\.
.
.TP
\fB\-n\fR, \fB\-\-max\-count\fR
Print only a specified number of commits\.
.
.SS "\fBmigrate\fR [\fI\-\-force\fR] \fIinstalled_formula\fR [\.\.\.]"
Migrate renamed packages to new names, where \fIformula\fR are old names of packages\.
.
.TP
\fB\-f\fR, \fB\-\-force\fR
Treat installed \fIformula\fR and provided \fIformula\fR as if they are from the same taps and migrate them anyway\.
.
.SS "\fBmissing\fR [\fI\-\-hide\fR\fB=\fR] [\fIformula\fR \.\.\.]"
Check the given \fIformula\fR kegs for missing dependencies\. If no \fIformula\fR are provided, check all kegs\. Will exit with a non\-zero status if any kegs are found to be missing dependencies\.
.
.TP
\fB\-\-hide\fR
Act as if none of the specified \fIhidden\fR are installed\. \fIhidden\fR should be a comma\-separated list of formulae\.
.
.SS "\fBoptions\fR [\fIoptions\fR] [\fIformula\fR \.\.\.]"
Show install options specific to \fIformula\fR\.
.
.TP
\fB\-\-compact\fR
Show all options on a single line separated by spaces\.
.
.TP
\fB\-\-installed\fR
Show options for formulae that are currently installed\.
.
.TP
\fB\-\-all\fR
Show options for all available formulae\.
.
.TP
\fB\-\-command\fR
Show options for the specified \fIcommand\fR\.
.
.SS "\fBoutdated\fR [\fIoptions\fR] [\fIformula\fR|\fIcask\fR \.\.\.]"
List installed casks and formulae that have an updated version available\. By default, version information is displayed in interactive shells, and suppressed otherwise\.
.
.TP
\fB\-q\fR, \fB\-\-quiet\fR
List only the names of outdated kegs (takes precedence over \fB\-\-verbose\fR)\.
.
.TP
\fB\-v\fR, \fB\-\-verbose\fR
Include detailed version information\.
.
.TP
\fB\-\-formula\fR
List only outdated formulae\.
.
.TP
\fB\-\-cask\fR
List only outdated casks\.
.
.TP
\fB\-\-json\fR
Print output in JSON format\. There are two versions: \fBv1\fR and \fBv2\fR\. \fBv1\fR is deprecated and is currently the default if no version is specified\. \fBv2\fR prints outdated formulae and casks\.
.
.TP
\fB\-\-fetch\-HEAD\fR
Fetch the upstream repository to detect if the HEAD installation of the formula is outdated\. Otherwise, the repository\'s HEAD will only be checked for updates when a new stable or development version has been released\.
.
.TP
\fB\-\-greedy\fR
Print outdated casks with \fBauto_updates\fR or \fBversion :latest\fR\.
.
.SS "\fBpin\fR \fIinstalled_formula\fR [\.\.\.]"
Pin the specified \fIformula\fR, preventing them from being upgraded when issuing the \fBbrew upgrade\fR \fIformula\fR command\. See also \fBunpin\fR\.
.
.SS "\fBpostinstall\fR \fIinstalled_formula\fR [\.\.\.]"
Rerun the post\-install steps for \fIformula\fR\.
.
.SS "\fBreadall\fR [\fI\-\-aliases\fR] [\fI\-\-syntax\fR] [\fItap\fR \.\.\.]"
Import all items from the specified \fItap\fR, or from all installed taps if none is provided\. This can be useful for debugging issues across all items when making significant changes to \fBformula\.rb\fR, testing the performance of loading all items or checking if any current formulae/casks have Ruby issues\.
.
.TP
\fB\-\-aliases\fR
Verify any alias symlinks in each tap\.
.
.TP
\fB\-\-syntax\fR
Syntax\-check all of Homebrew\'s Ruby files (if no \fB<tap>\fR is passed)\.
.
.SS "\fBreinstall\fR [\fIoptions\fR] \fIformula\fR|\fIcask\fR [\.\.\.]"
Uninstall and then reinstall a \fIformula\fR or \fIcask\fR using the same options it was originally installed with, plus any appended options specific to a \fIformula\fR\.
.
.P
If \fBHOMEBREW_BUILD_FROM_SOURCE\fR is set, then both \fIformula\fR and any dependencies reinstalled while reinstalling it are built from source even if bottles are available for them, regardless of whether \fB\-\-build\-from\-source\fR was passed or not\.
.
.P
Unless \fBHOMEBREW_NO_INSTALL_CLEANUP\fR is set, \fBbrew cleanup\fR will then be run for the reinstalled formulae or, every 30 days, for all formulae\.
.
.TP
\fB\-d\fR, \fB\-\-debug\fR
If brewing fails, open an interactive debugging session with access to IRB or a shell inside the temporary build directory\.
.
.TP
\fB\-f\fR, \fB\-\-force\fR
Install without checking for previously installed keg\-only or non\-migrated versions\.
.
.TP
\fB\-v\fR, \fB\-\-verbose\fR
Print the verification and postinstall steps\.
.
.TP
\fB\-\-formula\fR
Treat all named arguments as formulae\.
.
.TP
\fB\-s\fR, \fB\-\-build\-from\-source\fR
Compile \fIformula\fR from source even if a bottle is available\.
.
.TP
\fB\-i\fR, \fB\-\-interactive\fR
Download and patch \fIformula\fR, then open a shell\. This allows the user to run \fB\./configure \-\-help\fR and otherwise determine how to turn the software package into a Homebrew package\.
.
.TP
\fB\-\-force\-bottle\fR
Install from a bottle if it exists for the current or newest version of macOS, even if it would not normally be used for installation\.
.
.TP
\fB\-\-keep\-tmp\fR

Library/Homebrew/brew.rb
unless ENV["HOMEBREW_BREW_FILE"]
  raise "HOMEBREW_BREW_FILE was not exported! Please call bin/brew directly!"
end

std_trap = trap("INT") { exit! 130 } # no backtrace thanks

# check ruby version before requiring any modules.
RUBY_X, RUBY_Y, = RUBY_VERSION.split(".").map(&:to_i)
if RUBY_X < 2 || (RUBY_X == 2 && RUBY_Y < 3)
  raise "Homebrew must be run under Ruby 2.3! You're running #{RUBY_VERSION}."
end

# Also define here so we can rescue regardless of location.
class MissingEnvironmentVariables < RuntimeError; end

begin
  require_relative "global"
rescue MissingEnvironmentVariables => e
  raise e if ENV["HOMEBREW_MISSING_ENV_RETRY"]

  if ENV["HOMEBREW_DEVELOPER"]
    $stderr.puts <<~EOS
      Warning: #{e.message}
      Retrying with `exec #{ENV["HOMEBREW_BREW_FILE"]}`!
    EOS
  end

  ENV["HOMEBREW_MISSING_ENV_RETRY"] = "1"
  exec ENV["HOMEBREW_BREW_FILE"], *ARGV
end

begin
  trap("INT", std_trap) # restore default CTRL-C handler

  empty_argv = ARGV.empty?
  help_flag_list = %w[-h --help --usage -?]
  help_flag = !ENV["HOMEBREW_HELP"].nil?
  cmd = nil

  ARGV.dup.each_with_index do |arg, i|
    break if help_flag && cmd

    if arg == "help" && !cmd
      # Command-style help: `help <cmd>` is fine, but `<cmd> help` is not.
      help_flag = true
    elsif !cmd && !help_flag_list.include?(arg)
      cmd = ARGV.delete_at(i)
    end
  end

  path = PATH.new(ENV["PATH"])
  homebrew_path = PATH.new(ENV["HOMEBREW_PATH"])

  # Add SCM wrappers.
  path.prepend(HOMEBREW_SHIMS_PATH/"scm")
  homebrew_path.prepend(HOMEBREW_SHIMS_PATH/"scm")

  ENV["PATH"] = path

  if cmd
    internal_cmd = require? HOMEBREW_LIBRARY_PATH/"cmd"/cmd

    unless internal_cmd
      internal_dev_cmd = require? HOMEBREW_LIBRARY_PATH/"dev-cmd"/cmd
      internal_cmd = internal_dev_cmd
      if internal_dev_cmd && !ARGV.homebrew_developer?
        if (HOMEBREW_REPOSITORY/".git/config").exist?
          system "git", "config", "--file=#{HOMEBREW_REPOSITORY}/.git/config",
                                  "--replace-all", "homebrew.devcmdrun", "true"
        end
        ENV["HOMEBREW_DEV_CMD_RUN"] = "1"
      end
    end
  end

  unless internal_cmd
    # Add contributed commands to PATH before checking.
    homebrew_path.append(Tap.cmd_directories)

    # External commands expect a normal PATH
    ENV["PATH"] = homebrew_path
  end

  # Usage instructions should be displayed if and only if one of:
  # - a help flag is passed AND a command is matched
  # - a help flag is passed AND there is no command specified
  # - no arguments are passed
  # - if cmd is Cask, let Cask handle the help command instead
  if (empty_argv || help_flag) && cmd != "cask"
    require "help"
    Homebrew::Help.help cmd, empty_argv: empty_argv
    # `Homebrew.help` never returns, except for external/unknown commands.
  end

  # Uninstall old brew-cask if it's still around; we just use the tap now.
  if cmd == "cask" && (HOMEBREW_CELLAR/"brew-cask").exist?
    system(HOMEBREW_BREW_FILE, "uninstall", "--force", "brew-cask")
  end

  if internal_cmd
    Homebrew.send cmd.to_s.tr("-", "_").downcase
  elsif which "brew-#{cmd}"
    %w[CACHE LIBRARY_PATH].each do |env|
      ENV["HOMEBREW_#{env}"] = Object.const_get("HOMEBREW_#{env}").to_s
    end
    exec "brew-#{cmd}", *ARGV
  elsif (path = which("brew-#{cmd}.rb")) && require?(path)
    exit Homebrew.failed? ? 1 : 0
  else
    possible_tap = OFFICIAL_CMD_TAPS.find { |_, cmds| cmds.include?(cmd) }
    possible_tap = Tap.fetch(possible_tap.first) if possible_tap

    odie "Unknown command: #{cmd}" if !possible_tap || possible_tap.installed?

    brew_uid = HOMEBREW_BREW_FILE.stat.uid
    tap_commands = []
    if Process.uid.zero? && !brew_uid.zero?
      tap_commands += %W[/usr/bin/sudo -u ##{brew_uid}]
    end
    # Unset HOMEBREW_HELP to avoid confusing the tap
    ENV.delete("HOMEBREW_HELP") if help_flag
    tap_commands += %W[#{HOMEBREW_BREW_FILE} tap #{possible_tap}]
    safe_system(*tap_commands)
    ENV["HOMEBREW_HELP"] = "1" if help_flag
    exec HOMEBREW_BREW_FILE, cmd, *ARGV
  end
rescue UsageError => e
  require "help"
  Homebrew::Help.help cmd, usage_error: e.message
rescue SystemExit => e
  onoe "Kernel.exit" if ARGV.debug? && !e.success?
  $stderr.puts e.backtrace if ARGV.debug?
  raise
rescue Interrupt
  $stderr.puts # seemingly a newline is typical
  exit 130
rescue BuildError => e
  Utils::Analytics.report_build_error(e)
  e.dump
  exit 1
rescue RuntimeError, SystemCallError => e
  raise if e.message.empty?

  onoe e
  $stderr.puts e.backtrace if ARGV.debug?
  exit 1
rescue MethodDeprecatedError => e
  onoe e
  if e.issues_url
    $stderr.puts "If reporting this issue please do so at (not Homebrew/brew or Homebrew/core):"
    $stderr.puts "  #{Formatter.url(e.issues_url)}"
  end
  $stderr.puts e.backtrace if ARGV.debug?
  exit 1
rescue Exception => e # rubocop:disable Lint/RescueException
  onoe e
  if internal_cmd && defined?(OS::ISSUES_URL) &&
     !ENV["HOMEBREW_NO_AUTO_UPDATE"]
    $stderr.puts "#{Tty.bold}Please report this bug:#{Tty.reset}"
    $stderr.puts "  #{Formatter.url(OS::ISSUES_URL)}"
  end
  $stderr.puts e.backtrace
  exit 1
else
  exit 1 if Homebrew.failed?
end
